alfresco-rest-client/alfresco-rest-client-spring/src/main/java/eu/xenit/alfresco/restapi/client/spring/NodesRestApiSpringClient.java

```diff
@@ -0,0 +1,148 @@
+import eu.xenit.alfresco.restapi.client.spi.query.NodeCreateBody;
+import eu.xenit.alfresco.restapi.client.spi.query.PaginationQueryParameters;
+import eu.xenit.alfresco.restapi.client.spi.query.QueryParameters;
+import java.net.URI;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpClientErrorException.BadRequest;
+import org.springframework.web.client.HttpClientErrorException.Conflict;
+import org.springframework.web.client.HttpClientErrorException.Forbidden;
+import org.springframework.web.client.HttpClientErrorException.NotFound;
+import org.springframework.web.client.HttpClientErrorException.Unauthorized;
+import org.springframework.web.util.UriComponentsBuilder;
+
+public class NodesRestApiSpringClient extends RestApiSpringClient implements NodesRestApiClient {
+
+    public NodesRestApiSpringClient(AlfrescoRestProperties alfrescoRestProperties) {
+        super(alfrescoRestProperties);
+    }
+
+    @Override
+    protected String getApiBasePath() {
+        return "/nodes";
+    }
+
+    @Override
+    public void delete(String nodeId, DeleteNodeQueryParameters queryParameters) {
+        URI uri = withQueryParameters(UriComponentsBuilder
+                .fromHttpUrl(url).path("/" + nodeId), queryParameters)
+                .build().toUri();
+
+        RequestEntity<Void> requestEntity = RequestEntity.delete(uri).build();
+        execute(nodeId, requestEntity);
+    }
+
+    @Override
+    public NodeEntry get(String nodeId, GetNodeQueryParameters queryParameters) {
+        URI uri = withQueryParameters(UriComponentsBuilder
+                .fromHttpUrl(url).path("/" + nodeId), queryParameters)
+                .build().toUri();
+
+        RequestEntity<Void> requestEntity = RequestEntity.get(uri).build();
+        return execute(nodeId, requestEntity, NodeEntry.class);
+    }
+
+    @Override
+    public NodeList getChildren(String nodeId, PaginationQueryParameters paginationQueryParameters,
+            GetNodeQueryParameters nodeQueryParameters) {
+        URI uri = children(nodeId, paginationQueryParameters, nodeQueryParameters);
+
+        RequestEntity<Void> requestEntity = RequestEntity.get(uri).build();
+        return execute(nodeId, requestEntity, NodeList.class);
+    }
+
+    @Override
+    public NodeEntry createChild(String nodeId, NodeCreateBody nodeCreateBody,
+            CreateNodeQueryParameters queryParameters) {
+        RequestEntity<NodeCreateBody> requestEntity = RequestEntity.post(children(nodeId, queryParameters))
+                .body(nodeCreateBody);
+
+        return execute(nodeId, requestEntity, NodeEntry.class);
+    }
+
+    @Override
+    public NodeList getSources(String nodeId, GetAssociationsQueryParameters queryParameters) {
+        RequestEntity<Void> requestEntity = RequestEntity.get(uri(nodeId, "/sources", queryParameters)).build();
```

Comment:
can you align this with the method for getting targets?
`uri(nodeId, "/sources", queryParameters)` <-> `targets(nodeId, queryParameters)`

Reply:
Only used once, is this worth the overhead?